modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,389 @@
+		if ( $this->get_attempts() >= $max_attempts ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sets the status of the current job.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $status Status to set for current job.
+	 * @return bool Returns true if status is updated, false otherwise.
+	 */
+	public function set_status( $status ) {
+		$valid_statuses = array(
+			self::JOB_STATUS_COMPLETE,
+			self::JOB_STATUS_FAILED,
+			self::JOB_STATUS_QUEUED,
+			self::JOB_STATUS_RUNNING,
+			self::JOB_STATUS_PARTIAL,
+		);
+
+		if ( in_array( $status, $valid_statuses, true ) ) {
+			update_term_meta( $this->id, self::META_KEY_JOB_STATUS, $status );
+
+			// If job is complete, set the timestamp at which it was completed.
+			if ( self::JOB_STATUS_COMPLETE === $status ) {
+				update_term_meta( $this->id, self::META_KEY_JOB_COMPLETED_AT, time() );
+			}
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Records the error for the current process run.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param WP_Error $error Error instance.
+	 *
+	 * @todo Implement the error recording activity.
+	 *
+	 * @return void
+	 */
+	public function set_error( WP_Error $error ) {
+		$job_failure_data = $error->get_error_data( 'perflab_job_failure' );
+
+		if ( ! empty( $job_failure_data ) ) {
+			$this->set_status( self::JOB_STATUS_FAILED );
+
+			update_term_meta( $this->id, self::META_KEY_JOB_ERRORS, $job_failure_data );
+			update_term_meta( $this->id, self::META_KEY_JOB_ATTEMPTS, ( $this->get_attempts() + 1 ) );
+		}
+	}
+
+	/**
+	 * Retrieves the job status from its meta information.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Job status.
+	 */
+	public function get_status() {
+		return (string) get_term_meta( $this->id, self::META_KEY_JOB_STATUS, true );
+	}
+
+	/**
+	 * Returns the number of attempts executed for a job.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return int
+	 */
+	public function get_attempts() {
+		$attempts = get_term_meta( $this->id, self::META_KEY_JOB_ATTEMPTS, true );
+
+		return (int) $attempts;
+	}
+
+	/**
+	 * Set the start time of job.
+	 * It tells at what point of time the job has been started.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $time Timestamp (in seconds) at which job has been started.
+	 *
+	 * @return void
+	 */
+	public function lock( $time = null ) {
+		$time = empty( $time ) ? time() : $time;
+		update_term_meta( $this->id, self::META_KEY_JOB_LOCK, $time );
+		$this->set_status( self::JOB_STATUS_RUNNING );
+	}
+
+	/**
+	 * Get the timestamp (in seconds) when the job was started.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return int Start time of the job.
+	 */
+	public function get_start_time() {
+		$time = get_term_meta( $this->id, self::META_KEY_JOB_LOCK, true );
+
+		return absint( $time );
+	}
+
+	/**
+	 * Unlocks the process.
+	 *
+	 * Mark job status as partial as the unlock implies that job has run
+	 * atleast partially.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return void
```

Comment:
Any return types of `void` should not be documented, per WPCS.
```suggestion
	 * @since n.e.x.t
```